feat(incidents): implement models and client for /incidents/secrets/ (#115)
GitGuardian/py-gitguardian: changelog.d/20240823_170420_fnareoh_scrt_4759_optimize_files_to_scan_on_merge_commits.md

### Added

> - Added the models and client for to retrieve a secret incident https://api.gitguardian.com/docs#tag/Secret-Incidents/operation/retrieve-incidents

Reviewer: ```suggestion
- Added `GGClient.retrieve_secret_incident()` to retrieve the incident associated with a secret (see https://api.gitguardian.com/docs#tag/Secret-Incidents/operation/retrieve-incidents)
```
